mazerunner-scenarios/src/main/cpp/bugsnags.cpp

```diff
@@ -0,0 +1,205 @@
+//#include <bugsnag.h>
+#include <jni.h>
+#include <stdlib.h>
+
+extern "C" {
+//JNIEXPORT void JNICALL
+//Java_com_bugsnag_android_mazerunner_scenarios_CXXCustomMetadataNativeNotifyScenario_activate(JNIEnv *env,
+//                                                                         jobject instance) {
+//  bugsnag_notify_env(env, (char *)"Twitter Overdose",
+//                     (char *)"Turn off the internet and go outside", BSG_SEVERITY_INFO);
+//}
+//
+//JNIEXPORT void JNICALL
+//Java_com_bugsnag_android_mazerunner_scenarios_CXXAutoContextScenario_activate(JNIEnv *env,
+//                                                                         jobject instance) {
+//  bugsnag_notify_env(env, (char *)"Hello hello",
+//                     (char *)"This is a new world", BSG_SEVERITY_INFO);
+//}
+
+JNIEXPORT int JNICALL
+Java_com_bugsnag_android_mazerunner_scenarios_CXXStartSessionScenario_crash(JNIEnv *env,
+                                                                            jobject instance,
+                                                                            jint value) {
+    int x = 22;
+    if (x > 0)
+        __builtin_trap();
+    return 338;
+}
+
+JNIEXPORT int JNICALL
+Java_com_bugsnag_android_mazerunner_scenarios_CXXStopSessionScenario_crash(JNIEnv *env,
+                                                                           jobject instance,
+                                                                           jint value) {
+    int x = 22552;
+    if (x > 0)
+        __builtin_trap();
+    return 555;
+}
+
+JNIEXPORT int JNICALL
+Java_com_bugsnag_android_mazerunner_scenarios_CXXUpdateContextCrashScenario_crash(JNIEnv *env,
+                                                                         jobject instance,
+                                                                         jint value) {
+  if (value > 0)
+    __builtin_trap();
+  return 12067 / value;
+}
+
+JNIEXPORT int JNICALL
+Java_com_bugsnag_android_mazerunner_scenarios_CXXSessionInfoCrashScenario_crash(JNIEnv *env,
+                                                                         jobject instance,
+                                                                         jint value) {
+  if (value > 0)
+    __builtin_trap();
+  return 48206 / value;
+}
+
+JNIEXPORT int JNICALL
+Java_com_bugsnag_android_mazerunner_scenarios_CXXExtraordinaryLongStringScenario_crash(JNIEnv *env,
+                                                                         jobject instance,
+                                                                         jint value) {
+  if (value > 0)
+    __builtin_trap();
+  return 12062 / value;
+}
+
+//JNIEXPORT void JNICALL
```

Comment:
Calls which use bugsnag's C API have been commented out for now, and are compiled from the individual `features` and `tests/features` folders instead.